src/commands/branch.rs

```diff
@@ -293,15 +294,39 @@ fn cmd_branch_list(
 
     let mut formatter = ui.stdout_formatter();
     let formatter = formatter.as_mut();
-    for (name, branch_target) in repo.view().branches() {
+
+    let mut all_branches = repo.view().branches().clone();
+    for (branch_name, git_tracking_target) in git_tracking_branches(repo.view()) {
+        let branch_target = all_branches
+            .entry(branch_name.to_owned())
+            .or_insert(BranchTarget {
+                local_target: None,
+                remote_targets: BTreeMap::default(),
+            });
+        // TODO: This is not currently tested
+        // TODO: Use `BTreeMap::try_insert` once that's stable. That will allow removing
+        // the unwrap()
```

Comment:
Until then, maybe it's better to use `if branch_target.remote_targets.contains_key("git") { <print warning>?; }`